CONTRIBUTING.md

```diff
@@ -6,29 +6,322 @@ This page is intended for contributors to the registry and packages.
+
+    Generated packages are stored by default in the `dev/packages/beats` directory. Generally, the import process
+    updates all of the integrations, so don't be surprised if you notice updates to multiple integrations, including
+    the one you're currently working on (e.g. `dev/packages/beats/foobarbaz-0.0.1`). You can either commit this changes
+    or leave them for later.
+
+6. Copy the package output for your integration (e.g. `dev/packages/beats/foobarbaz-0.0.1`) to the _alpha_ directory and
+    raise the version manually: `dev/packages/alpha/foobarbaz-0.0.2`.
+
+### Fine-tune the integration
+
+#### Motivation
+
+Most of migration work has been done by the `import-beats` script, but there're tasks that require developer's
+interaction.
+
+It may happen that your integration misses a screenshot or an icon, it's a good moment to add missing resources to
+Beats/Kibana repositories and re-import the integration (idempotent). 
+
+#### Checklist
+
+The order of action items on the checklist is advised to prevent the contributor from repeating some actions (fixing
+what's been already fixed, as the script has overridden part of it).
+
+1. Add icon if missing.
+
+    The integration icons are presented in different places in Kibana, hence it's better to define custom icons to make
+    the UI easier to navigate.
+
+    As the `import-beats` script looks for icons in Kibana and EUI repositories, add an icon to the first one the same
+    way as for tutorial resources (Kibana directory: `src/legacy/core_plugins/kibana/public/home/tutorial_resources/logos/`).
+
+2. Add screenshot if missing.
+
+    The Kibana Integration Manager shows screenshots related with the integration. Screenshots present Kibana
+    dashboards visualizing the metric/log data.
+
+    The `import-beats` script finds references to screenshots mentioned in `_meta/docs.asciidoc` and copies image files
+    from the Beats directories:
+    * `metricbeat/docs/images`
+    * `filebeat/docs/images`
+
+3. Improve/correct spelling product names.
+
+    The correct spelling of product names simply makes better impression. The `import-beats` scripts uses the `fields.yml`
+    file as the source of the correct spelling (`title` property), e.g. Mysql - MySQL, Nginx - NGINX, Aws - AWS.
+
+    Keep in mind that this step requires reimporting package contents.
+
+4. Write README template file for the integration.
+
+    The README template is used to render the final README file including exported fields. The template should be placed
+    in the `dev/beats/import-beats-resources/docs/<integration-name>/docs/README.md`.
+
+    Review the MySQL docs template to see how to use template functions (e.g. `{{fields "dataset-name"}}`)
+
+5. Review fields file and exported fields in docs.
+
+    The fields files (`package-fields.yml`, `fields.yml` and `ecs.yml`) in the package were created from original
```

Comment:
Could you add an additional note here that the sum of all the fields.yml for a dataset should only contain the fields that are really used by the dataset? For example not just have all the ecs fields that exist?

Reply:
Right, done.